snuba/datasets/entities/discover.py

```diff
@@ -455,8 +455,8 @@ def callback_func(
                         sentry_sdk.capture_message(
                             "Non matching Discover result - different length",
                             level="warning",
-                            tags={"referrer": referrer},
-                            extra={
+                            tags={
+                                "referrer": referrer,
                                 "query": query.get_body(),
```

Comment:
This is unlikely to work. tags are not structured and are limited in length it will be very hard to read anything from the tag. 
Afaik this is the place where logger messages are transformed in Sentry events, so since extra data works for logger, you should be able to find here how to produce an event with extra data
https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/integrations/logging.py#L161

Reply:
Fixed the extras key (it's supposed to be extras with an 's')